src/main/scala/com/fulcrumgenomics/fasta/SortSequenceDictionary.scala

```diff
@@ -0,0 +1,98 @@
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.fasta.SequenceMetadata.Keys
+import com.fulcrumgenomics.sopt._
+import com.fulcrumgenomics.util.Io
+import enumeratum.EnumEntry
+
+import scala.collection.immutable.IndexedSeq
+import scala.collection.mutable.ListBuffer
+
+
+@clp(description =
+  """
+    |Sorts a sequence dictionary file in the order of another sequence dictionary.
+    |
+    |The inputs are to two `*.dict` files.  One to be sorted, and the other to provide the order for the sorting.
+    |
+    |The output will be a "sequence dictionary", which is a valid SAM file, containing the version header line and one
+    |line per contig.  The fields of the entries in this dictionary will be the same as in input, but in the order of
+    |sortDictionary.
+  """,
+  group = ClpGroups.Fasta)
+class SortSequenceDictionary
+(@arg(flag='i', doc="Input sequence dictionary file to be sorted.") val input: FilePath,
+ @arg(flag='d', doc="Input sequence dictionary file containing contigs in the desired sort order.") val sortDictionary: PathToSequenceDictionary,
+ @arg(flag='o', doc="Output sequence dictionary file.") val output: PathToSequenceDictionary,
+ @arg(doc="Append contigs that have no matching contig in the existing dictionary to the end of the output dictionary. If false, such contigs will not be output") val appendMissingContigs: Boolean = true,
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(input)
+  Io.assertReadable(sortDictionary)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+      val input_dict = SequenceDictionary(input)
+      val sort_order_dict = SequenceDictionary(sortDictionary)
+
+      val new_order: Builder[SequenceMetadata, IndexedSeq[SequenceMetadata]] = IndexedSeq.newBuilder[SequenceMetadata]
+      var new_index_counter: Int = 0
+
+      sort_order_dict.foreach { sort_meta => 
+          input_dict.get(sort_meta.name) match {
+              case Some(in_meta) => {
+                  new_order += in_meta.copy(index=new_index_counter)
+                  new_index_counter += 1
+              }
+              case None => {
+                  logger.info(s"Contig '${sort_meta.name}' corresponded to no contig in input dictionary, skipping")
+              }
+          }
+      }
+
+      val tmp_dict = SequenceDictionary(new_order.result().toSeq:_*)
+
+      input_dict.foreach { in_meta => 
+          input_dict.get(in_meta.name) match {
+              case None => {
+                  val skipBehavior = if (appendMissingContigs) {"Appending."} else {"Skipping."}
+                  logger.info(s"Contig '${in_meta.name}' was not found in sort order dictionary ${skipBehavior}")
+                  if(appendMissingContigs) {
+                      new_order += in_meta.copy(index=new_index_counter)
+                      new_index_counter += 1
+                  }
+              }
+              case Some(_) => {}
+          }
```

Comment:
1. use `contains` since an `Option` is just a collection of zero or one things.  If you have an empty `None` or `Some` match, likely there's a collection method (e.g. foreach, getOrElse, contains) that you can use instead.
2. No need for curly braces in one line if/else.
3. No need curly braces in cases in match statements
4. Non 
```suggestion
          if (!input_dict.contains(in_meta.name)) {
                val skipBehavior = if (appendMissingContigs) "appending." else "skipping."
                logger.info(s"Contig '${in_meta.name}' was not found in sort order dictionary: ${skipBehavior}")
                if (appendMissingContigs) {
                    new_order += in_meta.copy(index=new_index_counter)
                    new_index_counter += 1
                }
          }
```